Lib/test/test_capi/test_mem.py

```diff
@@ -75,12 +75,20 @@ def check_malloc_without_gil(self, code):
     def test_pymem_malloc_without_gil(self):
         # Debug hooks must raise an error if PyMem_Malloc() is called
         # without holding the GIL
+        if support.check_sanitizer(thread=True) and sys.platform == 'darwin':
+            # See: gh-120696
+            raise unittest.SkipTest("this test will hang on macOS with TSAN")
```

Comment:
If this is hanging in faulthandler, that probably means the test is **crashing**.

Reply:
Faulthandler hanging is definitely not great, but even if faulthandler is fixed I expect there will still be a problem with this test.

Reply:
These tests are just test the mallocator without the GIL, and intented to be crashed, so I think they are ok for now?

Reply:
Oh, that makes sense -- I missed that.